File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/shared/collectionitem.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class CollectionItem:
    r"""CollectionItem
    Metadata about an item that is part of a collection
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    added_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('added_time'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    media_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('media_type') }})
    

File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/shared/collectionitem.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// CollectionItem
/** 
 * Metadata about an item that is part of a collection
**/
export class CollectionItem extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=added_time" })
  addedTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=media_type" })
  mediaType?: string;
}
